cub/cub/device/device_scan.cuh

```diff
@@ -1770,14 +1779,14 @@ struct DeviceScan
     ValuesOutputIteratorT d_values_out,
     ScanOpT scan_op,
     InitValueT init_value,
-    int num_items,
+    NumItemsT num_items,
     EqualityOpT equality_op = EqualityOpT(),
     cudaStream_t stream     = 0)
   {
     CUB_DETAIL_NVTX_RANGE_SCOPE_IF(d_temp_storage, "cub::DeviceScan::ExclusiveScanByKey");
 
-    // Signed integer type for global offsets
-    using OffsetT = int;
+    // Unsigned integer type for global offsets
+    using OffsetT = detail::choose_offset_t<NumItemsT>;
```

Comment:
question: this currently means that CUB tests cover unsigned offsets, while Thrust uses `DispatchScanByKey` directly with signed `int32` and `int64` types. Should we update Thrust to use choose offset type?

Reply:
> Should we update Thrust to use choose offset type?

Yes, absolutely. I updated `thrust::scan_by_key` to also use unsigned offset types.